### ISSUE TITLE
Compilation with 'make' fails :/

### ISSUE THREAD

**Issue by mlliarm** [ei14/nre#1]
Hello,

Great project (what a crazy crazy idea!), hence the star.

Tried compiling the code but `make` failed:

```
milia@Newton:~/Documents/dev/c/nre$ ls
config.h  LICENSE.txt  Makefile  nre.c  nre.h  README.md
milia@Newton:~/Documents/dev/c/nre$ make
gcc nre.c -o nre -lm -lpthread -Ofast
nre.c:7:22: error: initializer element is not constant
 const int COUNT012 = COUNT01 + COUNT2;
                      ^~~~~~~
Makefile:2: recipe for target 'main' failed
make: *** [main] Error 1
```

Here's my `gcc -v`, as you might locate something there that is missing.
```
milia@Newton:~/Documents/dev/c/nre$ gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/7/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 7.5.0-3ubuntu1~18.04' --with-bugurl=file:///usr/share/doc/gcc-7/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++ --prefix=/usr --with-gcc-major-version-only --program-suffix=-7 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-bootstrap --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-libmpx --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 7.5.0 (Ubuntu 7.5.0-3ubuntu1~18.04)
```

Are there any C libraries I should have already pre-installed in order for this to be compiled?

Thanks !

**Comment by ei14** [comment]
That's odd; I could not replicate the error with gcc version 11.1.0. I pushed a change just now that modifies the definition of `COUNT012` to not rely on any other variables; I think that may fix the issue.

**Comment by mlliarm** [comment]
Hey @ei14 ,

Thanks for the response.

Actually I've upgraded by gcc version lately to `gcc version 8.4.0 (Ubuntu 8.4.0-1ubuntu1~18.04)` and now it compiles just fine in the previous version of `nre.c`, before the fix you did.

I also compiled the new version of the `nre.c` file with my old gcc-7.5.0 compiler, and works as well. So it seems that the fix did the trick.

Summing up: 
- in order to use the older version of the code without your fix I had to use a newer version of gcc (gcc-8.4.0)
- the new version of the code runs with the 'problematic' gcc-7.5.0 compiler as well as with gcc-8.4.0.

Closing the issue since it's fixed. Thanks !
